src/EntityGraphQL/Compiler/Util/ExpressionUtil.cs

```diff
@@ -413,9 +429,9 @@ public static Expression MakeSelectWithDynamicType(string fieldDescription, Para
                 return baseExp;
 
             // get a list of distinct types asked for in the query (fragments for interfaces)
-            var validTypes = fieldExpressions.Values.OfType<MemberExpression>()
```

Comment:
support conditional fields by recursively visiting expression to find type instead